core/primitives/src/syncing.rs

```diff
@@ -188,6 +191,34 @@ impl ShardStateSyncResponseV1 {
     }
 }
 
+#[derive(Serialize, BorshSerialize, BorshDeserialize, Eq, PartialEq, Debug, Clone)]
+pub struct EpochSyncFinalizationResponse {
+    pub cur_epoch_header: BlockHeader,
+    pub prev_epoch_headers: Vec<BlockHeader>,
```

Comment:
is this bounded in size?

Reply:
Cannot be longer than the length of the Epoch.